### PR TITLE
OOP school library: decorate a class & set up associations

### PR DESCRIPTION
**Pull request by felixodette**
* Implement composition (as an example of the design pattern).
* Set up associations between classes and objects.
* Run a program using the command line.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @felixodette,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have used the wrong git flow. You are expected to merge from feature branch to develop branch, not develop to main

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative './nameable'
+
+class BaseDecorator
+  def initialize(nameable)
+    @nameable = nameable
```
Comment (medaminedev66) [inline base_decorator.rb:5]: [OPTIONAL] Getters and Setters for the instance variable `@nameable` should be present.

**Review by medaminedev66 — APPROVED** [review]
Hi @felixodette 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- There are no Linters errors
- Your code is clean

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by felixodette** [comment]
Thanks for the quick review. I'll have a look at the ```@nameable``` getters and setters shortly.

**Pull request merged**